src/main/scala/mesosphere/marathon/core/launchqueue/impl/ReviveOffersActor.scala

```diff
@@ -0,0 +1,89 @@
+package mesosphere.marathon
+package core.launchqueue.impl
+
+import akka.actor.{Actor, Props, Stash, Status}
+import akka.pattern.pipe
+import akka.event.LoggingReceive
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.{Sink, Source}
+import akka.{Done, NotUsed}
+import com.typesafe.scalalogging.StrictLogging
+import mesosphere.marathon.core.instance.update.InstanceChangeOrSnapshot
+import mesosphere.marathon.core.task.tracker.InstanceTracker
+import mesosphere.marathon.metrics.{Counter, Metrics}
+
+import scala.concurrent.duration._
+
+sealed trait Op
+case object Revive extends Op
+case object Suppress extends Op
+
+class ReviveOffersActor(
+    metrics: Metrics,
+    minReviveOffersInterval: FiniteDuration,
+    instanceUpdates: InstanceTracker.InstanceUpdates,
+    rateLimiterUpdates: Source[RateLimiter.DelayUpdate, NotUsed],
+    driverHolder: MarathonSchedulerDriverHolder,
+    enableSuppress: Boolean) extends Actor with Stash with StrictLogging {
+
+  private[this] val reviveCountMetric: Counter = metrics.counter("mesos.calls.revive")
+  private[this] val suppressCountMetric: Counter = metrics.counter("mesos.calls.suppress")
+
+  import context.dispatcher
+  implicit val mat = ActorMaterializer()(context)
+
+  override def preStart(): Unit = {
+    super.preStart()
+
+    val delayedConfigRefs: Source[ReviveOffersStreamLogic.DelayedStatus, NotUsed] =
+      rateLimiterUpdates.via(ReviveOffersStreamLogic.activelyDelayedRefs)
+
+    val flattenedInstanceUpdates: Source[InstanceChangeOrSnapshot, NotUsed] = instanceUpdates.flatMapConcat {
+      case (snapshot, updates) =>
+        Source.single[InstanceChangeOrSnapshot](snapshot).concat(updates)
+    }
+
+    val suppressReviveFlow = ReviveOffersStreamLogic.suppressAndReviveFlow(
+      minReviveOffersInterval = minReviveOffersInterval,
+      enableSuppress = enableSuppress)
+
+    val done = flattenedInstanceUpdates.map(Left(_))
+      .merge(delayedConfigRefs.map(Right(_)))
+      .via(suppressReviveFlow)
+      .runWith(Sink.foreach {
+        case Revive =>
+          reviveCountMetric.increment()
```

Comment:
Ok, so this is potentially an easy solution into the debuggability of the whole suppress/revive topic - what if we'd already introduced per role metrics like:
```
marathon.roles.slave_public.suppressed = 1
marathon.roles.slave_public.revived = 0
```
to show whether or not we're now in suppress or revive mode. This is also nicely extendable later when we'll go multi-role. wdyt /cc @timcharper @jeschkies ?

Reply:
That's a pretty good idea; we could improve it even further and count the number of seconds we're in each corresponding state. That would potentially be more useful and give the same info by comparing two values.

Reply:
Oh yeah! This is even better!

Reply:
On the second thought - having to meter time for the roles in each state might create too much complexity. I would be fine with the binary metric too.